auraed/src/ebpf/tracepoint_programs/tracepoint_program.rs

```diff
@@ -0,0 +1,130 @@
+use super::PerfEventListener;
+use aya::programs::ProgramError;
+use aya::{
+    maps::perf::AsyncPerfEventArray,
+    programs::TracePoint,
+    util::{nr_cpus, online_cpus},
+    Bpf,
+};
+use bytes::BytesMut;
+use log::trace;
+use procfs::page_size;
+use std::mem::size_of;
+use tokio::sync::broadcast;
+use tracing::error;
+
+/// Size (in pages) for the circular per-CPU buffers that BPF perfbuf creates.
+const PER_CPU_BUFFER_SIZE_IN_PAGES: usize = 2;
+
+pub trait TracepointProgram<T: Clone + Send + 'static> {
+    const PROGRAM_NAME: &'static str;
+    const CATEGORY: &'static str;
+    const EVENT: &'static str;
+    const PERF_BUFFER: &'static str;
+
+    fn load_and_attach(
+        bpf: &mut Bpf,
+    ) -> Result<PerfEventListener<T>, anyhow::Error> {
+        trace!("Loading eBPF program: {}", Self::PROGRAM_NAME);
+
+        // Load the eBPF TracePoint program
+        let program: &mut TracePoint = bpf
+            .program_mut(Self::PROGRAM_NAME)
+            .ok_or_else(|| anyhow::anyhow!("failed to get eBPF program"))?
+            .try_into()?;
+
+        // Load the program
+        match program.load() {
+            Ok(_) => Ok(()),
+            Err(ProgramError::AlreadyLoaded) => Ok(()),
```

Comment:
Seemed like it would be ok to ignore this error

Reply:
Let's add a log line, so we know when something fishy is going on. We shouldn't be trying to load the same program twice.